packages/block-editor/src/components/list-view/private-list-view.js

```diff
@@ -0,0 +1,270 @@
+import { __experimentalTreeGrid as TreeGrid } from '@wordpress/components';
+import { AsyncModeProvider, useSelect } from '@wordpress/data';
+import deprecated from '@wordpress/deprecated';
+import {
+	useCallback,
+	useEffect,
+	useMemo,
+	useRef,
+	useReducer,
+	forwardRef,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+import { BlockSettingsDropdown } from '../block-settings-menu/block-settings-dropdown';
+
+import ListViewBranch from './branch';
+import { ListViewContext } from './context';
+import ListViewDropIndicator from './drop-indicator';
+import useBlockSelection from './use-block-selection';
+import useListViewClientIds from './use-list-view-client-ids';
+import useListViewDropZone from './use-list-view-drop-zone';
+import useListViewExpandSelectedItem from './use-list-view-expand-selected-item';
+
+const expanded = ( state, action ) => {
+	if ( Array.isArray( action.clientIds ) ) {
+		return {
+			...state,
+			...action.clientIds.reduce(
+				( newState, id ) => ( {
+					...newState,
+					[ id ]: action.type === 'expand',
+				} ),
+				{}
+			),
+		};
+	}
+	return state;
+};
+
+export const BLOCK_LIST_ITEM_HEIGHT = 36;
+
+/** @typedef {import('react').ComponentType} ComponentType */
+/** @typedef {import('react').Ref<HTMLElement>} Ref */
+
+/**
+ * Show a hierarchical list of blocks.
+ *
+ * @param {Object}         props                         Components props.
+ * @param {string}         props.id                      An HTML element id for the root element of ListView.
+ * @param {Array}          props.blocks                  _deprecated_ Custom subset of block client IDs to be used instead of the default hierarchy.
+ * @param {?boolean}       props.showBlockMovers         Flag to enable block movers. Defaults to `false`.
+ * @param {?boolean}       props.isExpanded              Flag to determine whether nested levels are expanded by default. Defaults to `false`.
+ * @param {?boolean}       props.showAppender            Flag to show or hide the block appender. Defaults to `false`.
+ * @param {?ComponentType} props.blockSettingsMenu       Optional more menu substitution. Defaults to the standard `BlockSettingsDropdown` component.
+ * @param {string}         props.rootClientId            The client id of the root block from which we determine the blocks to show in the list.
+ * @param {string}         props.description             Optional accessible description for the tree grid component.
+ * @param {?Function}      props.onSelect                Optional callback to be invoked when a block is selected. Receives the block object that was selected.
+ * @param {Function}       props.renderAdditionalBlockUI Function that renders additional block content UI.
+ * @param {Ref}            ref                           Forwarded ref
+ */
+const PrivateListView = forwardRef( function ListViewComponent(
```

Comment:
Apologies if this has already been discussed somewhere, but I'm wondering why this component is being moved to another file? It creates quite a big diff, which means that we could lose some useful history in the `ListViewComponent`. Is it possible to keep `ListViewComponent` in `index.js`?

Reply:
I moved it back in [3fe3bf9](https://github.com/WordPress/gutenberg/pull/49417/commits/3fe3bf99b7507df47a2c24f3aee04ec2322f94c2)